torch/linalg/__init__.py

```diff
@@ -1438,49 +1642,57 @@
 
-.. note::
-          Backpropagation is not supported for ``mode='r'``. Use ``mode='reduced'`` instead.
+The returned decomposition is a named tuple `(Q, R)`.
+If :attr:`input` is a batch of matrices, then `Q`, `R` are also batched with the same
+batch dimensions.
 
-          Backpropagation is also not supported if the first
-          :math:`\min(input.size(-1), input.size(-2))` columns of any matrix
-          in :attr:`input` are not linearly independent. While no error will
-          be thrown when this occurs the values of the "gradient" produced may
-          be anything. This behavior may change in the future.
+The parameter :attr:`mode` controls the shape of the output. If :attr:`input` has shape
+`(*, m, n)`, denoting `k = min(m, n)`
 
-.. note:: This function uses LAPACK for CPU inputs and MAGMA for CUDA inputs,
-          and may produce different (valid) decompositions on different device types
-          or different platforms.
+- :attr:`mode` `= 'reduced'` (default): Returns `(Q, R)` of shapes `(*, m, k)`, `(*, k, n)` respectively.
+- :attr:`mode` `= 'complete'`: Returns `(Q, R)` of shapes `(*, m, m)`, `(*, m, n)` respectively.
+- :attr:`mode` `= 'r'`: Computes only `R`. Returns `(Q, R)` with `Q` empty and `R` of shape `(*, k, n)`.
 
-Args:
-    input (Tensor): the input tensor of size :math:`(*, m, n)` where `*` is zero or more
-                batch dimensions consisting of matrices of dimension :math:`m \times n`.
-    mode (str, optional): if `k = min(m, n)` then:
+Differences with `numpy.linalg.qr`:
+
+- :attr:`mode` `= 'raw'` is not implemented.
+- Unlike `numpy.linalg.qr`, this function always returns a tuple of two tensors.
+  When :attr:`mode` `= 'r'`, the `Q` tensor is an empty tensor.
+  This behavior may change in a future PyTorch release.
 
-          * ``'reduced'`` : returns `(Q, R)` with dimensions (m, k), (k, n) (default)
+.. note:: The elements in the diagonal of `R` are not necessarily positive.
 
-          * ``'complete'``: returns `(Q, R)` with dimensions (m, m), (m, n)
+.. note:: :attr:`mode` `= 'r'` does not support backpropagation. Use :attr:`mode` `= 'reduced'` instead.
 
-          * ``'r'``: computes only `R`; returns `(Q, R)` where `Q` is empty and `R` has dimensions (k, n)
+.. note:: This function uses LAPACK and MAGMA for CPU and CUDA inputs respectively.
+
+.. warning:: The QR decomposition is only locally unique when the input matrix has
+             independent columns. If this is not the case, different platforms, like NumPy,
+             or inputs on different devices, may produce different valid decompositions.
+
+.. warning:: Backpropagation is only supported if the first `k = min(m, n)` columns
+             of every matrix in :attr:`input` are linearly independent.
+             If this condition is not met, no error will be thrown, but the gradient produced
+             may be anything.
```

Comment:
"may be anything" is a bit unclear. Maybe "If this condition is not met, no error will be thrown, but the gradient produced is not correct"?